.mvn/gradle-enterprise-custom-user-data.groovy

```diff
@@ -96,3 +96,20 @@ if (System.env.GITHUB_ACTIONS) {
     }
 }
 
+// Check current Maven version and Maven Wrapper version  alignment
+def runtimeInfo = (org.apache.maven.rtinfo.RuntimeInformation) session.lookup("org.apache.maven.rtinfo.RuntimeInformation")
+def currentMavenVersion = runtimeInfo?.getMavenVersion()
+Properties mavenWrapperProperties = new Properties()
+File mavenWrapperPropertiesFile = new File(".mvn/wrapper/maven-wrapper.properties")
+if(mavenWrapperPropertiesFile.exists()) {
+    mavenWrapperPropertiesFile.withInputStream {
+        mavenWrapperProperties.load(it)
+    }
+    if(regexp = mavenWrapperProperties."distributionUrl" =~ /.*\/apache-maven-(.*)-bin\.zip/) {
```

Comment:
This will work for Quarkus but we if the maven wrapper is distributed from a custom URL will this still work?

Reply:
I am parsing here a standard URL, this could not work with a custom URL indeed.
I am adding a comment to describe what is expected.